lib/site.taxonomy.term.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Term class
+ */
+
+class Term {
+	/**
+	 * Create a Term instance
+	 *
+	 * @param {String} siteId - site id
+	 * @param {String} taxonomy - taxonomy type
+	 * @param {WPCOM} wpcom - wpcom instance
+	 * @return {Null} null
+	 */
+	constructor( siteId, taxonomy, term, wpcom ) {
+		if ( ! siteId ) {
+			throw new Error( '`siteId` is not correctly defined' );
+		}
+
+		if ( ! taxonomy ) {
+			throw new Error( '`taxonomy` is not correctly defined' );
+		}
+
+		if ( ! ( this instanceof Term ) ) {
+			return new Term( siteId, taxonomy, term, wpcom );
+		}
+
+		this.wpcom = wpcom;
+
+		this._siteId = encodeURIComponent( siteId );
+		this._taxonomy = encodeURIComponent( taxonomy );
+		this._term = encodeURIComponent( term );
+		this._taxonomyPath = `/sites/${this._siteId}/taxonomies/${this._taxonomy}/terms`;
+	}
+
+	/**
+	 * Get Term details
+	 *
+	 * @param {Object} [query] - query parameters
+ 	 * @param {Function} fn - callback function
+ 	 * @return {Function} request handler
```

Comment:
I'm afraid we are not returning the request handler anymore.
![image](https://cloud.githubusercontent.com/assets/77539/14456669/670dfbc6-007c-11e6-92ad-af477d732752.png)

Now we are returning a Promise. https://github.com/Automattic/wpcom.js/blob/master/lib/site.post.subscriber.js#L45